### ISSUE TITLE
X509 tests intermittently failing

### ISSUE THREAD

**Issue by CIPop** [Azure/azure-iot-sdk-csharp#171]
See: https://github.com/Azure/azure-iot-sdk-csharp/pull/170

@ryan-z-johnson I believe you are already looking into this. Once it's fixed, please remove the TODOs from the code.

The following tests are intermittently failing:

(updated per @jasmineymlo:)
```

Failed                Microsoft.Azure.Devices.E2ETests.X509E2ETests.X509_DeviceSendSingleMessage_Amqp
Failed                Microsoft.Azure.Devices.E2ETests.X509E2ETests.X509_DeviceSendSingleMessage_AmqpWs
Failed                Microsoft.Azure.Devices.E2ETests.X509E2ETests.X509_DeviceReceiveSingleMessage_Amqp 
Failed                Microsoft.Azure.Devices.E2ETests.X509E2ETests.X509_DeviceReceiveSingleMessage_AmqpWs
Parameter name: Key
```

```
Test method Microsoft.Azure.Devices.E2ETests.X509E2ETests.X509_DeviceSendSingleMessage_AmqpWs threw exception: 
System.ArgumentException: '' is not a valid Base64 encoded string.
Parameter name: Key
```

```
Microsoft.Azure.Devices.Client.DeviceClient.<>c__DisplayClass64_0.<ApplyTimeout>b__2(Task t) in D:\jenkins\workspace\win-csharp-vs2017\device\Microsoft.Azure.Devices.Client\DeviceClient.cs: line 814
System.Threading.Tasks.ContinuationTaskFromTask.InnerInvoke()
System.Threading.Tasks.Task.Execute()
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.Azure.Devices.E2ETests.X509E2ETests.<SendSingleMessageX509>d__23.MoveNext() in D:\jenkins\workspace\win-csharp-vs2017\e2e\Microsoft.Azure.Devices.E2ETests\X509E2ETests.cs: line 153
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Microsoft.Azure.Devices.E2ETests.X509E2ETests.<SendSingleMessageX509>d__23.MoveNext() in D:\jenkins\workspace\win-csharp-vs2017\e2e\Microsoft.Azure.Devices.E2ETests\X509E2ETests.cs: line 177
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.Azure.Devices.E2ETests.X509E2ETests.<X509_DeviceSendSingleMessage_AmqpWs>d__14.MoveNext() in D:\jenkins\workspace\win-csharp-vs2017\e2e\Microsoft.Azure.Devices.E2ETests\X509E2ETests.cs: line 73
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.GetResult()
```

/cc @jasmineymlo

**Comment by ryan-z-johnson** [comment]
Are you seeing this on your local machine?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by CIPop** [comment]
> Are you seeing this on your local machine?

I've just hit this in the CI.

**Comment by jasmineymlo** [comment]
@ryan-z-johnson I saw it on local, random as well.

**Comment by CIPop** [comment]
Failed for `Microsoft.Azure.Devices.E2ETests.X509E2ETests.X509_DeviceSendSingleMessage_Amqp` as well.

```
Test method Microsoft.Azure.Devices.E2ETests.X509E2ETests.X509_DeviceSendSingleMessage_Amqp threw exception: 
System.ArgumentException: '' is not a valid Base64 encoded string.
Parameter name: Key
```

```
Microsoft.Azure.Devices.Client.DeviceClient.<>c__DisplayClass64_0.<ApplyTimeout>b__2(Task t) in D:\jenkins\workspace\win-csharp-vs2017\device\Microsoft.Azure.Devices.Client\DeviceClient.cs: line 814
System.Threading.Tasks.ContinuationTaskFromTask.InnerInvoke()
System.Threading.Tasks.Task.Execute()
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.Azure.Devices.E2ETests.X509E2ETests.<SendSingleMessageX509>d__23.MoveNext() in D:\jenkins\workspace\win-csharp-vs2017\e2e\Microsoft.Azure.Devices.E2ETests\X509E2ETests.cs: line 153
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
Microsoft.Azure.Devices.E2ETests.X509E2ETests.<SendSingleMessageX509>d__23.MoveNext() in D:\jenkins\workspace\win-csharp-vs2017\e2e\Microsoft.Azure.Devices.E2ETests\X509E2ETests.cs: line 177
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.Azure.Devices.E2ETests.X509E2ETests.<X509_DeviceSendSingleMessage_Amqp>d__13.MoveNext() in D:\jenkins\workspace\win-csharp-vs2017\e2e\Microsoft.Azure.Devices.E2ETests\X509E2ETests.cs: line 66
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.GetResult()
```

**Comment by CIPop** [comment]
New failures today. I will disable all X509/AMQP tests for now against this bug.

**Comment by jenoola** [comment]
Fixed in 1.20.0

**Comment by az-iot-builder-01** [comment]
@CIPop, @jenoola, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
